examples/centralized_threshold.rs

```diff
@@ -0,0 +1,172 @@
+//! Centralized Telescope example with BLS multi-signature
+use crate::aggregate_signature::helpers::{
+    collect_valid_signatures, get_commitment, validate_signatures,
+};
+use crate::aggregate_signature::registration::Registration;
+use crate::aggregate_signature::signature::IndividualSignature;
+use crate::aggregate_signature::signer::{RegisteredSigner, Signer};
+use alba::centralized_telescope::params::Params;
+use alba::centralized_telescope::proof::Proof;
+use alba::centralized_telescope::CentralizedTelescope;
+use rand_chacha::ChaCha20Rng;
+use rand_core::{RngCore, SeedableRng};
+
+mod aggregate_signature;
+const DATA_LENGTH: usize = 48;
+pub(crate) type Element = [u8; DATA_LENGTH];
+
+#[derive(Debug, Clone)]
+pub(crate) struct AlbaThresholdSignature {
+    /// Centralized telescope proof
+    pub(crate) proof: Proof,
+    /// Registration indices of the element sequence signers
+    pub(crate) indices: Vec<usize>,
+    /// Commitment `Hash(checksum || msg)`
+    pub(crate) commitment: Vec<u8>,
+}
+
+impl AlbaThresholdSignature {
+    /// Create AlbaThresholdSignature. Validate and collect signatures in byte representation.
+    /// Create Alba proof and extract indices of proof elements.
+    /// Return proof, commitment, and indices.
+    fn prove<const N: usize>(
+        params: &Params,
+        signature_list: &[IndividualSignature],
+        registration: &Registration,
+        msg: &[u8],
+        set_size: usize,
+    ) -> Option<Self> {
+        // Ensure that the commitment can be created
+        let Some(checksum) = &registration.checksum else {
+            println!("Error: Registration is not closed.");
+            return None;
+        };
+
+        let commitment = get_commitment::<N>(checksum, msg).to_vec();
+
+        // Collect valid individual signatures' byte representation into a hashmap
+        let valid_signatures = collect_valid_signatures::<N>(signature_list, registration, msg);
+
+        // Check if there are enough valid signatures
+        if valid_signatures.len() < set_size {
+            println!(
+                "Error: Not enough signatures! Expected at least {}, but got {}.",
+                set_size,
+                valid_signatures.len()
+            );
+            return None;
+        }
+
+        // Collect the byte representation of valid signatures into a Vec
+        let prover_set: Vec<Element> = valid_signatures.keys().copied().collect();
```

Comment:
Potentially stupid question: could we generate the proof directly on the indices?

Reply:
We might (?)